cacti/test_cacti.py

```diff
@@ -0,0 +1,126 @@
+# (C) Datadog, Inc. 2010-2016
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+# stdlib
+import logging
+import os
+import shutil
+from nose.plugins.attrib import attr
+
+# project
+from shared.test.common import AgentCheckTest, Fixtures
+
+log = logging.getLogger()
+
+CONFIG = """
+init_config:
+
+instances:
+    -   mysql_host: localhost
+        mysql_user: root
+        rrd_path:   /tmp/cacti_test/rrds
+        rrd_whitelist: %s
+""" % Fixtures.file('whitelist.txt')
+
+@attr(requires='cacti')
+class TestCacti(AgentCheckTest):
+
+    CHECK_NAME = 'cacti'
+
+    CACTI_CONFIG = {
+        'mysql_host': 'localhost',
+        'mysql_user': 'root',
+        'rrd_path':   '/tmp/cacti_test/rrds',
+        'rrd_whitelist': "%s" % Fixtures.file('whitelist.txt')
+    }
+
+    def setUp(self):
+        self.tmp_dir = '/tmp/cacti_test'
+        self.rrd_dir = os.path.join(os.path.dirname(__file__), "cacti")
+
+        # Create our temporary RRD path, if needed
+        try:
+            os.mkdir(self.tmp_dir)
+        except Exception:
+            # Ignore, directory already exists
+            pass
+
+    def tearDown(self):
+        # Clean up the temp directory
+        shutil.rmtree(self.tmp_dir)
+
+    def _copy_rrds(self, xml_dir):
+        if os.access("/usr/bin/rrdtool", os.R_OK | os.X_OK):
```

Comment:
I don't know if it's safe to assume`/usr/bin/rrdtool` will be available on travis/circleci? Furthermore, how do we know it'll be available in our contributors environments? The test should be self-contained. It should provide everything required to run it, and then clean after itself. See the comment in the rakefile.

If this fails, we will return `False` and the test will silently complete without running. See my comment below.